src/pal/tests/palsuite/file_io/FILECanonicalizePath/FILECanonicalizePath.c

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+//
+
+/*=====================================================================
+**
+** Source:  FILECanonicalizePath.c (test 1)
+**
+** Purpose: Tests the PAL implementation of the FILECanonicalizePath function.
+**
+**
+**===================================================================*/
+
+#include <palsuite.h>
+
+extern void FILECanonicalizePath(LPSTR lpUnixPath);
+
+void TestCase(LPSTR input, LPSTR expectedOutput);
+
+int __cdecl main(int argc, char *argv[])
+{
+    if (PAL_Initialize(argc,argv) != 0)
+    {
+        return FAIL;
+    }
+
+    // Case 01: /<name> should not change
+    TestCase("/Test", "/Test");
+
+    // Case 02: /<name>/<name2> should not change
+    TestCase("/Test/Foo", "/Test/Foo");
+
+    // Case 03: // transforms to /
+    TestCase("//", "/");
+  
+    // Case 04: /./ transforms to /
+    TestCase("/./", "/");
+    
+    // Case 05: /<name>/../ transforms to /
+    TestCase("/Test/../", "/");
+        
+    // Case 06: /Test/Foo/.. transforms to /Test
+    TestCase("/Test/Foo/..", "/Test");
+        
+    // Case 07: /Test/.. transforms to /
+    TestCase("/Test/..", "/");
+        
+    // Case 08: /. transforms to /
+    TestCase("/.", "/");
```

Comment:
Looks like the implementation special cases "/.." as well, so a test case for that would be good, too.